pex/resolver.py

```diff
@@ -839,7 +842,9 @@ def _check_install(installed_distributions):
                     )
                 else:
                     installed_dist = installed_requirement_dist.distribution
-                    if installed_dist not in requirement:
+                    if not requirement.specifier.contains(
+                        installed_dist.version, prereleases=self._allow_prereleases
```

Comment:
This threading of `--pre` broke a Pants test. The underlying old code for
```python
installed_dist not in requirement
```
is in `pkg_resources.Requirement.__contains__` which always allows prereleases:
https://github.com/pantsbuild/pex/blob/be0d71230c2ccf3ae53c5051d09ece1859728bb0/pex/vendor/_vendored/setuptools/pkg_resources/__init__.py#L3160-L3170